### PR TITLE
OKTA-232895 - Add React Native instructions to Guide: Sign users in to your mobile apps

### PR DESCRIPTION
**Pull request by IldarAbdullin-okta**
### Resolves:

* [OKTA-232895](https://oktainc.atlassian.net/browse/OKTA-232895)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Call `isAuthenticated` function if you want to check whether the user is signed-in:
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md]: * Call _the_ `isAuthenticated` function

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+```javascript
+const result = await isAuthenticated();
+if (result.authenticated !== this.state.authenticated) {
+    this.setState({authenticated: result.authenticated});
+}
+```
+
+Implementation in iOS and Android native modules checks if the client have an access or ID token. Then the user is considered authenticated and this call will return true. This does not check the validity of the access token which could be expired or revoked. See <GuideLink link="../stay-signed-in">Keep the User Signed In</GuideLink> to understand how to get a fresh access token.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md]: * client _has_ an access 
or
* _clients_ have an access

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/configuremid.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Call `createConfig` function to create a configured client on the native modules. Resolves true if successfully configures a client. Note: `requireHardwareBackedKeyStore` is a configurable setting only on Android devices. If you're testing on Android emulators, set this field to false
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/configuremid.md]: * "if _it_ successfully configures"
* "this field to `false`."

Notes should be formatted on new lines:

```
> **Note:**
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+The async `signIn` method will automatically redirect users to your Okta organziation for authentication. It will emit an event once a user successfully signs in. Make sure your event listeners are mounted and unmounted. Note: on iOS there isn't a onCancelled event. If the sign in process is cancelled, onError will be triggered.
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md]: * "Okta _organization_ "
* "_an_ `onCancelled` event"
* "`onError` will be triggered"

Notes should be formatted on new lines:

```
> **Note:**
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/use-access-token/react-native/usetoken.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+Get the access token using the `getAccessToken` method and prepare a http header`:
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/use-access-token/react-native/usetoken.md]: * "_an HTTP_ header"

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/next-steps/react-native/samples.md [hunk 6]
```diff
@@ -0,0 +1 @@
+We have sample application available in our [React Native samples repo](https://github.com/okta/samples-js-react-native) on GitHub.
```
Comment (albertchen-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/next-steps/react-native/samples.md]: We have *a* sample application...

Comment (albertchen-okta) [reply packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/next-steps/react-native/samples.md]: or "our"

**Comment by IldarAbdullin-okta** [comment]
@jakubvul-okta , @albertchen-okta , thanks for comments! PR updated

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Call the `isAuthenticated` function if you want to check whether the user is signed-in:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md]: no need to hyphenate signed-in

Call the `isAuthenticated` function if you want to check whether the user is signed in:

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+
+```javascript
+const result = await isAuthenticated();
+if (result.authenticated !== this.state.authenticated) {
+    this.setState({authenticated: result.authenticated});
+}
+```
+
+Implementation in iOS and Android native modules checks if the client has an access or ID token. Then the user is considered authenticated and this call will return true. This does not check the validity of the access token which could be expired or revoked. See <GuideLink link="../stay-signed-in">Keep the User Signed In</GuideLink> to understand how to get a fresh access token.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/check-on-startup/react-native/checkfortoken.md]: 2 nits - removed "will" and added an s to returns
made does not a contraction for conversational tone

Implementation in iOS and Android native modules checks if the client has an access or ID token. Then the user is considered authenticated and this call returns true. This doesn't check the validity of the access token which could be expired or revoked. See <GuideLink link="../stay-signed-in">Keep the User Signed In</GuideLink> to understand how to get a fresh access token.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/configuremid.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+Call `createConfig` function to create a configured client on the native modules. Resolves true if it successfully configures a client.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/configuremid.md]: "the" `createConfig` function

Call the `createConfig` function to create a configured client on the native modules. Resolves true if it successfully configures a client.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+```bash
+$ npm install @okta/okta-react-native --save
+$ react-native link @okta/okta-react-native
+```
+
+### Install iOS OIDC SDK
+
+1. [**CocoaPods**](https://guides.cocoapods.org/using/getting-started.html)
+
+***React Native >= 0.60***: With React Native 0.60 pods are added to podfile automatically. Run `pod install` command to install dependecies:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: Couple of nits - comma before pds..."the" before `pod install`, dependencies typo (was missing the n)

***React Native >= 0.60***: With React Native 0.60, pods are added to the podfile automatically. Run the `pod install` command to install dependencies:

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+```
+cd ios
+pod install
+```
+***React Native < 0.60***: Make sure your `Podfile` looks like this:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: added "that"

***React Native < 0.60***: Make sure that your `Podfile` looks like this:

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+```
+Then run `pod install`.
+
+2. [**Carthage**](https://github.com/Carthage/Carthage)
+```
+github "okta/okta-oidc-ios" ~> 3.5.0
+```
+Then run `carthage update --platform iOS`.
+
+Open project settings and choose your application target. Then open `Build Phases` and add `OktaOidc.framework` from `ios/Carthage/Build/iOS` into `Embed Frameworks` section
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: "the" in front of `Embed Frameworks` and period at the end

Open project settings and choose your application target. Then open `Build Phases` and add `OktaOidc.framework` from `ios/Carthage/Build/iOS` into the `Embed Frameworks` section.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+### Install Android OIDC SDK
+
+1. Add this line to `android/build.gradle`, under `allprojects` -> `repositories`.
+```
+maven {
+url  "https://dl.bintray.com/okta/com.okta.android"
+}
+```
+
+2. Make sure your `minSdkVersion` is `19` in `android/build.gradle`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: added "that"

2. Make sure that your `minSdkVersion` is `19` in `android/build.gradle`.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+3. Add redirect scheme in `android/app/build.gradle`, under `android` -> `defaultConfig`:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: added "the"

Add the redirect scheme in `android/app/build.gradle`, under `android` -> `defaultConfig`:

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/define-callback/react-native/definescheme.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+<GuideLink link=”../ios”>Set up  iOS redirect scheme</GuideLink>
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/define-callback/react-native/definescheme.md]: removed extra space in front if iOS

<GuideLink link=”../ios”>Set up iOS redirect scheme</GuideLink>

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
+The async `signIn` method will automatically redirect users to your Okta organization for authentication. It will emit an event once a user successfully signs in. Make sure your event listeners are mounted and unmounted.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md]: removed "will" and added an s to redirect and emit, added "that" in front of your - Make sure that your..

The async `signIn` method automatically redirects users to your Okta organization for authentication. It emits an event once a user successfully signs in. Make sure that your event listeners are mounted and unmounted.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+**Note:** On iOS there isn't an `onCancelled` event. If the sign in process is cancelled, `onError` will be triggered.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/sign-in-page/react-native/signin.md]: hypenated sign-in, as it is describing process here and removed "will be" and replaced with "is"

**Note:** On iOS there isn't an `onCancelled` event. If the sign-in process is cancelled, `onError` is triggered.

**Comment by IldarAbdullin-okta** [comment]
Thanks you, @susanharper-okta! PR updated.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md [hunk 18]
```diff
@@ -32,7 +32,7 @@ github "okta/okta-oidc-ios" ~> 3.5.0
 ```
 Then run `carthage update --platform iOS`.
 
-Open project settings and choose your application target. Then open `Build Phases` and add `OktaOidc.framework` from `ios/Carthage/Build/iOS` into `Embed Frameworks` section
+Open project settings and choose your application target. Then open `Build Phases` and add `OktaOidc.framework` from `ios/Carthage/Build/iOS` into the `Embed Frameworks` section
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app/configure-packages/react-native/installsdk.md]: can you put a period at the end of this sentence please :)

Open project settings and choose your application target. Then open `Build Phases` and add `OktaOidc.framework` from `ios/Carthage/Build/iOS` into the `Embed Frameworks` section**.**

**Comment by jakubvul** [comment]
🚢

**Pull request merged**